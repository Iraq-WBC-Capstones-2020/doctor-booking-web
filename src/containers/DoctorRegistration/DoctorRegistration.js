import React, { useState } from 'react';
import { Stepper, Step } from 'react-form-stepper';
import { Container, Button, Form } from 'react-bootstrap';
import StepThree from '../../components/DoctorRegisterPage/StepThree/StepThree';
import StepOne from '../../components/DoctorRegisterPage/StepOne/StepOne';
import StepTwo from '../../components/DoctorRegisterPage/StepTwo/StepTwo';

function DoctorRegistration() {
  const [activeStep, setActiveStep] = useState(0);
  const handleNextStep = () => {
    if (activeStep <= 1) {
      setActiveStep(activeStep + 1);
    }
  };
  const handleBackStep = () => {
    if (activeStep >= 1) {
      setActiveStep(activeStep - 1);
    }
  };
  const stepStyleConfig = {
    activeBgColor: '#026fc2',
    completedBgColor: '#28a745',
  };
  return (
    <Container>
      <Form>
        <Stepper styleConfig={stepStyleConfig} activeStep={activeStep}>
          <Step label="Personal info" />
          <Step label="Career info" />
          <Step label="Clinic info" />
        </Stepper>
      </Form>
      {activeStep === 0 ? (
        <h1>
          <StepOne />
        </h1>
      ) : (
        ''
      )}
      {activeStep === 1 ? <StepTwo /> : ''}
      {activeStep === 2 ? <StepThree /> : ''}

      <Button
        className={activeStep === 0 ? 'disabled mr-3' : 'mr-3'}
        onClick={handleBackStep}
      >
        Back
      </Button>
      <Button onClick={handleNextStep}>Next</Button>
    </Container>
  );
}

export default DoctorRegistration;
