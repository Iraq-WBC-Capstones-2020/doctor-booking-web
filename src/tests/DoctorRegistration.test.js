import React from 'react';
import { render } from '@testing-library/react';
import DoctorRegistration from '../containers/DoctorRegistration/DoctorRegistration';
import renderer from 'react-test-renderer';
test('Doctor Registration is rendering ', () => {
  render(<DoctorRegistration />);
});

test('Doctor Registration renders correctly', () => {
  const tree = renderer.create(<DoctorRegistration />).toJSON();
  expect(tree).toMatchSnapshot();
});
