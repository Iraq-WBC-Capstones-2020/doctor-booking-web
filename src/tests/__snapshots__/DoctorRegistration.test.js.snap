// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doctor Registration renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="RFS-StepperContainer"
  >
    <div
      className="RFS-StepContainer"
    >
      <div
        className="RFS-StepMain"
      >
        <button
          className="RFS-StepCircle RFS-StepCircle-d12 RFS-StepButton active"
          disabled={false}
        >
          <span
            className="RFS-StepCircleContent RFS-StepCircleContent-d13 active"
          >
            1
          </span>
        </button>
        <div
          className="RFS-LabelContainer RFS-LabelContainer-d14"
        >
          <span
            className="RFS-Label RFS-Label-d15"
          >
            Personal info
          </span>
        </div>
      </div>
    </div>
    <div
      className="RFS-StepContainer"
    >
      <div
        className="RFS-ConnectorContainer RFS-ConnectorContainer-d4"
      >
        <span
          className="RFS-Connector RFS-Connector-d5"
        />
      </div>
      <div
        className="RFS-StepMain"
      >
        <button
          className="RFS-StepCircle RFS-StepCircle-d16 RFS-StepButton"
          disabled={true}
        >
          <span
            className="RFS-StepCircleContent RFS-StepCircleContent-d17"
          >
            2
          </span>
        </button>
        <div
          className="RFS-LabelContainer RFS-LabelContainer-d18"
        >
          <span
            className="RFS-Label RFS-Label-d19"
          >
            Career info
          </span>
        </div>
      </div>
    </div>
    <div
      className="RFS-StepContainer"
    >
      <div
        className="RFS-ConnectorContainer RFS-ConnectorContainer-d6"
      >
        <span
          className="RFS-Connector RFS-Connector-d7"
        />
      </div>
      <div
        className="RFS-StepMain"
      >
        <button
          className="RFS-StepCircle RFS-StepCircle-d20 RFS-StepButton"
          disabled={true}
        >
          <span
            className="RFS-StepCircleContent RFS-StepCircleContent-d21"
          >
            3
          </span>
        </button>
        <div
          className="RFS-LabelContainer RFS-LabelContainer-d22"
        >
          <span
            className="RFS-Label RFS-Label-d23"
          >
            Clinic info
          </span>
        </div>
      </div>
    </div>
  </div>
  <h1>
    step one
  </h1>
  
  
  <button
    className="disabled mr-3 btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Back
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Next
  </button>
</div>
`;
